src/tasks/node/NodeTaskHelper.ts

```diff
@@ -0,0 +1,212 @@
+        }
+
+        if (buildOptions.tag === undefined) {
+            buildOptions.tag = getDefaultImageName(packageName);
+        }
+
+        return buildOptions;
+    }
+
+    public async resolveDockerRunOptions(context: DockerRunTaskContext, runDefinition: NodeRunTaskDefinition): Promise<DockerRunOptions> {
+        const helperOptions = runDefinition.node || {};
+        const runOptions = runDefinition.dockerRun;
+
+        const packagePath = NodeTaskHelper.inferPackagePath(helperOptions && helperOptions.package, context.folder);
+        const packageName = await NodeTaskHelper.inferPackageName(packagePath);
+
+        if (runOptions.containerName === undefined) {
+            runOptions.containerName = getDefaultContainerName(packageName);
+        }
+
+        if (runOptions.image === undefined) {
+            runOptions.image = inferImageName(runDefinition, context, packageName);
+        }
+
+        if (helperOptions && helperOptions.enableDebugging) {
+            const inspectMode = helperOptions.inspectMode || 'default';
+            const inspectPort = helperOptions.inspectPort !== undefined ? helperOptions.inspectPort : 9229;
+
+            if (runOptions.command === undefined) {
+                runOptions.command = await NodeTaskHelper.inferCommand(packagePath, inspectMode, inspectPort);
+            }
+
+            if (runOptions.ports === undefined) {
+                runOptions.ports = [];
+            }
+
+            let inspectPortPublished = false;
+
+            // If not already defined, create a mapping for the inspect port between container and host...
+            if (runOptions.ports.find(port => port.containerPort === inspectPort) === undefined) {
+                runOptions.ports.push({
+                    containerPort: inspectPort,
+                    // TODO: Can this mapping be dynamic?
+                    hostPort: inspectPort
+                });
+
+                inspectPortPublished = true;
+            }
+
+            // NOTE: By default, if no ports are explicitly published and the options do not say otherwise, all ports will be published.
+            //       If we published the inspection port, and it was the only published port, that "publish all" behavior would unintentionally be disabled.
+            //       Hence, in that situation, we force "publish all" in addition to the inspection port.
+            if (runOptions.portsPublishAll === undefined && inspectPortPublished && runOptions.ports.length === 1) {
+                runOptions.portsPublishAll = true;
+            }
+        }
+
+        return runOptions;
+    }
+
+    public static inferPackagePath(packageFile: string | undefined, folder: WorkspaceFolder): string {
+        if (packageFile !== undefined) {
+            return resolveFilePath(packageFile, folder);
+        } else {
+            return path.join(folder.uri.fsPath, 'package.json');

```

Comment:
One common place for problems is a lot of users will put their project in a subfolder of their workspace. It's common for people that had a C# project in VS and then open it up in VS Code for the first time. It affects other languages, too - some people just like having the folder structure like that. 

Honestly you may or may not support those scenarios already - didn't fully investigate in this PR, but just wanted to make sure you considered that scenario

Reply:
Yes, I tried it out. By default VS makes such projects like this:

 - MySolution.sln
 - MyWebApp
   - MyWebApp.csproj
 - MyLibrary
   - MyLibrary.csproj

I tested with this config and it does work.